git/git.go

```diff
@@ -360,9 +458,16 @@ func DefaultRemote() (string, error) {
 	return "", errors.New("Unable to pick default remote, too ambiguous")
 }
 
-func UpdateIndex(file string) error {
-	_, err := subprocess.SimpleExec("git", "update-index", "-q", "--refresh", file)
-	return err
+func StartUpdateIndexFromStdin(output *bytes.Buffer) (io.WriteCloser, error) {
```

Comment:
In the same vein as my `git.HashObject()` comment, what do you think about changing this to accept an `io.Writer`? It'll give the caller some flexibility in how the output is captured.